### DISCUSSION TITLE
How to generate a range of “int“

### DISCUSSION THREAD

**Discussion by mNantern** [cue-lang/cue#1324] [Q&A] [answered]
Hello 👋.

I'm trying to convert [Spinnaker](https://spinnaker.io/) pipelines definition (which are giant json) to CUE and it's going great.

One think that I don't know how to do yet is generate a field called `refId` which is an int disguised in a string (??) and that represent the position of the object in a list:

```json
{
  "stages": [
    {
     …
      "refId": "1"
     …
    },
    {
      …
      "refId": "2"
      …
    },
    {
      …
      "refId": "3"
      …
    }
  ]
}
```

My questions are:
1. Is there a way to do it more cleanly with Cue?
2. I also need to generate a list of refId: `requisiteStageRefIds: ["21", "22", "23", "24", "25", "26", "27", "28"]`. Anything that I can do generate a range of int ?

Thank you 🙏

**Comment by seh** [comment]
There is [the `list.Range` function](https://pkg.go.dev/cuelang.org/go@v0.4.0/pkg/list#Range). You can then use [`strconv.FormatInt`](https://pkg.go.dev/cuelang.org/go@v0.4.0/pkg/strconv#FormatInt) to project the integers to strings.

**Reply by mNantern** [reply]
I should have looked at the function, thank you it's working great 🙏

**Comment by verdverm** [comment, marked as answer]
If you already have your objects in a list, you can unify the id into the object after string interpolation, in a list comprehension

```cue
refd: [ for i, obj in objs { obj & { refId: "\(i)" } }]
```

Use `refId: "\(i+1)"` if you want to start indexing at 1

https://cuelang.org/play/?id=ciMZWAwEWBK#cue@export@cue

**Reply by mNantern** [reply]
Thanks it's working now 🙏 : https://cuelang.org/play/?id=R0INrsecmqU#cue@export@json
